### DISCUSSION TITLE
Image (?/Markdown) support

### DISCUSSION THREAD

**Discussion by arminulrich** [spatie/laravel-ray#101] [Ideas]
Thank you for Ray - I love it, and after a few days I use it for literally everything. 🤩

One thing I miss that would make it even more awesome would be the ability to send an image URL to ray (and display it right away):

```php
ray()->image("https://placekitten.com/200/300")
```

or maybe in form of additional markdown support to get custom HTML into Ray:
```php
ray()->markdown("![https://placekitten.com/200/300](https://placekitten.com/200/300)")
```

...
or just some form of "rich" snippet, that would turn URLs into clickable links?
```php
ray()->richtext("https://placekitten.com/200/300"); 
```

what do you think?

EDIT:

maybe "HTML" support is be all we need, since everything else could be a ray() macro on top of that 

```php
Spatie\Ray\Ray::macro('image', function(string $image_url) {
  
    $this->sendHtml("<img src=" . $image_url . ">");
    
    return $this;
});

ray()->image("https://placekitten.com/200/300")
```

**Comment by patinthehat** [comment]
I think this could be useful. I'll submit a PR for `image()` to [spatie/ray](https://github.com/spatie/ray) and we'll see if it gets accepted.

**Comment by patinthehat** [comment]
Update: See [PR #170](https://github.com/spatie/ray/pull/170) in spatie/ray.

**Comment by arminulrich** [comment]
@patinthehat wow awesome thank you 👍
